### ISSUE TITLE
Make Swing changes for Mac

### ISSUE THREAD

**Issue by dcbriccetti** [opendatakit/briefcase#165]
Make the minor tweaks needed for a Swing application to work right on a Mac.

Related: https://github.com/opendatakit/validate/issues/30

**Comment by rockydcoder** [comment]
According to [this](https://stackoverflow.com/a/26080591/1508448) answer in SO, there are a number of ways to do this. But I want to know how this jar is packaged and made available to the end-user. @lognaturel , can you help please?

**Comment by yanokwa** [comment]
We run [gradle jar](https://github.com/opendatakit/briefcase/blob/master/build.gradle#L103) and the jar is what is distributed.

**Comment by rockydcoder** [comment]
So my assessment of this issue is this:

Setting the application name in a Java Swing application cannot be done programmatically. The solution to this was to use this line of code in the main class (`MainBriefcaseWindow`):

```java
System.setProperty("com.apple.mrj.application.apple.menu.about.name", "ODK Briefcase");
```

But that did not work (with Java 1.7/1.8; Mac OS - 10.11.6)

The easiest way to do this is to run the jar with this option:
```
java -Xdock:name="ODK Briefcase" -jar path-to-jar 
```

This instruction can be added in the instructions manual

Another way to do this is to add the application JAR to a Mac application bundle, as shown [here](https://stackoverflow.com/a/14485964/230513), an add the attribute to the Info.plist.

```
<key>JVMOptions</key>
<string>-Xdock:name=Name</string>
```

This way, instead of a jar, a `.dmg` file would be distributed.

Can I get some inputs from the repository maintainers?

**Comment by rockydcoder** [comment]
@lognaturel @yanokwa , can you please reply to my comment above?

**Comment by lognaturel** [comment]
I'm afraid I have no experience with this! I'm on a Mac and I've always let Java do whatever it does by default. @dcbriccetti @daveycrockett perhaps one of you has looked at these options recently and has good advice?

The Mac application bundle option is interesting to me. What are the possible downsides? Seems it might be awkward to run from the command line?

**Comment by dcbriccetti** [comment]
Since the old ways no longer work, I am in the dark.

**Comment by daveycrockett** [comment]
I tried the main two stackoverflow recommendations: 

- System.setProperty(), and
- moving main() to another class than the JFrame itself

I dumped all the System.getProperties() and didn't see anything mac-specific, so I'm fairly confident that this is indeed a JVM argument that can't be set at runtime.

Whether or not that's worth changing the distribution process to add a dmg build path for this platform? Don't know enough to weigh in concretely, but it does seem like it probably doesn't have a great impact-to-LOE ratio.
